com.unity.render-pipelines.universal/Editor/ShaderGUI/BaseShaderGUI.cs

```diff
@@ -368,16 +332,28 @@ protected static void DrawTileOffset(MaterialEditor materialEditor, MaterialProp
 
         public static void SetMaterialKeywords(Material material, Action<Material> shadingModelFunc = null, Action<Material> shaderFunc = null)
         {
-            // Clear all keywords for fresh start
-            material.shaderKeywords = null;
-
             // Setup blending - consistent across all Universal RP shaders
             SetupMaterialBlendMode(material);
 
             // Receive Shadows
             if (material.HasProperty("_ReceiveShadows"))
                 CoreUtils.SetKeyword(material, "_RECEIVE_SHADOWS_OFF", material.GetFloat("_ReceiveShadows") == 0.0f);
 
+            // Setup double sided GI
+            bool doubleSidedGI = (RenderFace)material.GetFloat("_Cull")  != RenderFace.Front;
+            if (doubleSidedGI != material.doubleSidedGI)
+                material.doubleSidedGI = doubleSidedGI;
+
+            // Temporary fix for lightmapping. TODO: to be replaced with attribute tag.
+            if (material.HasProperty("_MainTex"))
```

Comment:
There is [MainTexture] and [MainColor] attributes these days.
material.mainTexture, etc.
material.color
Should work these days. If not then it's a bug. A shader missing the attributes or the lookup is done incorrectly.

Reply:
Actually the attributes are there no the shader but the lightmapper doesn't use them, it accesses directly _MainTex and co
So we need to duplicate the properties value
Also note that this code was already present, i just moved it to this function to make sure it's executed when needed